src/key_store/mod.rs

```diff
@@ -0,0 +1,192 @@
+//! A storage solution for cryptographic key material.
+//!
+//! This module provides access to the `KeyStore` struct, which manages the
+//! storage of `CredentialBundle` and `KeyPackageBundle` instances. The
+//! development of this module is tracked in #337, which also includes a
+//! roadmap.
+//!
+//! The current key store enables the storage of `CredentialBundle` instances,
+//! and grants access to `CredentialBundle` references via the
+//! `SignaturePublicKey` of the corresponding `Credential`. The
+//! `CredentialBundle` instances are only accessible via reference, so that if a
+//! `KeyStore` is used in multiple groups, those groups can also share a
+//! `CredentialBundle`.
+//!
+//! The `KeyStore` also stores "init" `KeyPackageBundle` instances, i.e. the
+//! counterparts to the `KeyPackage` instances that are published so other
+//! parties can use them to add this party to groups. These `KeyPackageBundle`
+//! instances are removed from the `KeyStore` when they are requested via
+//! `take_key_package_bundle`. This is because each `ManagedGroup` owns the
+//! `KeyPackageBundle` in its leaf, so upon creation of the group, it needs to
```

Comment:
So this is what we do right now. But I'm not convinced yet that it's something we should be doing. This is why I'd like us to have an upgrade path from here and also describe it. I would expect us to deprecate the `take_` function here and instead introduce a `get_` function that is used when starting the client such that the `ManagedGroup` doesn't need to persist anything here.

Reply:
You're right. That was one of the questions that were left after our last discussion: Who should persist KeyPackages that are allocated to an individual group? Should it be the group (probably instructing the KeyStore to do it) or should it be the KeyStore directly?

Having written the answer to your other question, I now believe that it should be the KeyStore that holds (and persists) all the KeyPackageBundles and not the groups. This would allow it to keep track of expirations and active use of KeyPackageBundles and thus prevent the deletion of CredentialBundles if they are potentially still in use somewhere.

Reply:
>  I now believe that it should be the KeyStore that holds (and persists) all the KeyPackageBundles and not the groups

Yep, that's where I think we have to end up. But as I said I think we have this upgrade path. But we should also document where we (probably) want to go here.

Reply:
I've updated #337 to document this plan.